### ISSUE TITLE
create_parent fails when different columns are FK of same table

### ISSUE THREAD

**Issue by rdab** [keithf4/pg_partman#79]
hello,

I've a table where 2 of its columns are FK of another table. i.e:
``` sql
CREATE TABLE address (
eid serial not null primary key,
created_date timestamp without time zone not null,
blah text
);

CREATE TABLE deliverable(
eid serial not null,
created_date timestamp without time zone not null,
source_address integer REFERENCES address(eid) MATCH SIMPLE,  
dest_address integer REFERENCES address(eid) MATCH SIMPLE 
);

SELECT partman.create_parent('public.deliverable', 'created_date', 'time-static', 'monthly', NULL, 2, true, '2016-01-01', true, false)
```
`create_parent` function fails with error: 
`ERROR:  foreign key referenced-columns list must not contain duplicates`

Is this normal?
Thanks

**Comment by keithf4** [comment]
What version are you running? By the parameters to create_parent(), I'm assuming it's something in the 1.x series and not 2.x. The latest version (1.8.8) contains a fix for this. In the 2.x series, version 2.1.0 contains this fix. See the release notes at the top of this file
https://github.com/keithf4/pg_partman/blob/master/updates/pg_partman--1.8.7--1.8.8.sql

See the CHANGELOG entry for 2.1.0 for how to update to the latest 1.8.x version.
https://github.com/keithf4/pg_partman/blob/master/CHANGELOG.txt
```
keith=# \dx
                                               List of installed extensions
        Name        | Version |   Schema   |                                  Description                                  
--------------------+---------+------------+-------------------------------------------------------------------------------
 pg_partman         | 1.8.7   | partman    | Extension to manage partitioned tables by time or ID



keith=# CREATE TABLE public.address (
keith(# eid serial not null primary key,
keith(# created_date timestamp without time zone not null,
keith(# blah text
keith(# );
CREATE TABLE
Time: 12.321 ms

keith=# 
keith=# CREATE TABLE public.deliverable(
keith(# eid serial not null,
keith(# created_date timestamp without time zone not null,
keith(# source_address integer REFERENCES address(eid) MATCH SIMPLE,  
keith(# dest_address integer REFERENCES address(eid) MATCH SIMPLE 
keith(# );
CREATE TABLE
Time: 3.147 ms

keith=# SELECT partman.create_parent('public.deliverable', 'created_date', 'time-static', 'monthly', NULL, 2, true, '2016-01-01', true, false);
ERROR:  foreign key referenced-columns list must not contain duplicates
Time: 38.939 ms

keith=# alter extension pg_partman update to '1.8.8';
ALTER EXTENSION
Time: 24.915 ms

keith=# SELECT partman.create_parent('public.deliverable', 'created_date', 'time-static', 'monthly', NULL, 2, true, '2016-01-01', true, false);
 create_parent 
---------------
 t
(1 row)

Time: 57.021 ms
```

**Comment by keithf4** [comment]
If you are on PostgreSQL version 9.4, I would highly recommend upgrading to the 2.x series.

If you're not on 9.4, I would highly recommend looking into upgrading to allow management of this extension to be easier and get all the new features.

**Comment by rdab** [comment]
Thank you and excuse me. It's explained crystal clear how to upgrade from 1.8.7 to 1.8.8 in the README.md.

You're right. I'm using series 1.x, unfortunately my server runs 9.1 and is not my call to upgrade it. 

Thanks again, and keep doing such a good work.

**Comment by Xof** [comment]
We're getting this error on a 1.8.8 installation, while doing partman.create_parent. It's on PostgreSQL 9.2.18. Any suggestions on diagnosing it?

**Comment by keithf4** [comment]
I'll have to apologize and say I really can't provide any guidance on debugging v1.x anymore, nor any versions of PG that have gone out of support. I just don't have the time right now. 

Only thing I can recommend is to test this on a system with at least PG 9.4 and the latest version of pg_partman. If it's happening in the current version as well, I'd be happy to investigate there and then that might provide an answer as to how you can fix 1.8.8.

**Comment by Xof** [comment]
Fair enough... thanks!
